client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/Role.java

```diff
@@ -0,0 +1,271 @@
+        sb.append("], global[").append(manageApplicationPrivileges);
+        sb.append("], indicesPrivileges=[").append(Strings.collectionToCommaDelimitedString(indicesPrivileges));
+        sb.append("], applicationResourcePrivileges=[").append(Strings.collectionToCommaDelimitedString(applicationResourcePrivileges));
+        sb.append("], runAs=[").append(Strings.collectionToCommaDelimitedString(runAs));
+        sb.append("], metadata=[").append(metadata);
+        sb.append("]]");
+        return sb.toString();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (false == clusterPrivileges.isEmpty()) {
+            builder.field(CLUSTER.getPreferredName(), clusterPrivileges);
+        }
+        if (null != manageApplicationPrivileges) {
+            builder.field(GLOBAL.getPreferredName(), manageApplicationPrivileges);
+        }
+        if (false == indicesPrivileges.isEmpty()) {
+            builder.field(INDICES.getPreferredName(), indicesPrivileges);
+        }
+        if (false == applicationResourcePrivileges.isEmpty()) {
+            builder.field(APPLICATIONS.getPreferredName(), applicationResourcePrivileges);
+        }
+        if (false == runAs.isEmpty()) {
+            builder.field(RUN_AS.getPreferredName(), runAs);
+        }
+        if (false == metadata.isEmpty()) {
+            builder.field(METADATA.getPreferredName(), metadata);
+        }
+        return builder.endObject();
+    }
+
+    public static Role fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
+
+        private @Nullable Collection<ClusterPrivilege> clusterPrivileges = null;
+        private @Nullable ManageApplicationsPrivilege manageApplicationPrivileges = null;
+        private @Nullable Collection<IndicesPrivileges> indicesPrivileges = null;
+        private @Nullable Collection<ApplicationResourcePrivileges> applicationResourcePrivileges = null;
+        private @Nullable Collection<String> runAs = null;
+        private @Nullable Map<String, Object> metadata = null;
+
+        private Builder() {
+        }
+
+        public Builder clusterPrivileges(@Nullable ClusterPrivilege... clusterPrivileges) {
+            if (clusterPrivileges == null) {
+                // null is a no-op to be programmer friendly
+                return this;
+            }
+            return clusterPrivileges(Arrays.asList(clusterPrivileges));
+        }
+
+        public Builder clusterPrivileges(@Nullable Collection<ClusterPrivilege> clusterPrivileges) {
+            this.clusterPrivileges = clusterPrivileges;
+            return this;
+        }
+
+        public Builder manageApplicationPrivileges(@Nullable ManageApplicationsPrivilege manageApplicationPrivileges) {
```

Comment:
I definitely don't think we want a method that explicitly refers to `ManageApplicationsPrivilege`

Reply:
We can change the name or take a `Map` as an argument. We don't know how future "global" privileges will look like so I don't know what abstraction we should place here....

Reply:
This has been replaced.

The client is required to know to add `GlobalScopedPrivilege` with the `scope` to `manage`.